docs/_posts/2024-07-18-Run-your-pandas-workloads-on-cluster.md

```diff
@@ -0,0 +1,152 @@
+    import modin.pandas as pd
+    import ray
+    from modin.utils import execute # execute is just used to make sure all the asynchronous operations are complete as we benchmark.
+    ray.init(address="auto", logging_level="WARNING")
+    file_path = "/home/ray/data/big_yellow.csv"
+    df = pd.read_csv(file_path)
+    df = df.map(str)
+    result = pd.DataFrame()
+
+    payment_type_map = {
+        1: 'Credit Card',
+        2: 'Cash',
+        3: 'No Charge',
+        4: 'Dispute',
+        5: 'Unknown',
+        6: 'Voided Trip'
+    }
+
+    result["df_payment_type"] = df['payment_type'].map(payment_type_map)
+
+    def custom_function(row):
+        return str(row["passenger_count"]) + " passengers were picked up at " + str(row["tpep_pickup_datetime"])
+
+    result["description"] = df.apply(custom_function, axis=1)
+    execute(result)
+    </code></pre>
+
+  ```
+</details>
+
+
+### Results and discussion
+
+I observed that by increasing the number of cluster nodes, Modin could handle the workload effectively with 4 nodes or more. Significant performance improvements were evident as we scaled from 4 to 32 nodes. Given that Modin operates efficiently with 4 nodes and both Pandas and Modin struggle with fewer than 3 nodes, it suggests the workload demands nearly 4 times the memory (32GB) of a single t3.2xlarge instance.
+
+To provide a fair comparison of the performance gains offered by Modin over Pandas, we decided to benchmark the script on an instance with over 120 GB of memory. We chose an r6a.4xlarge instance, which boasts 128GiB of RAM.
+
+With Pandas, the script took almost 400 seconds to execute, which serves as our baseline.
+
+
+
+### Performance on a Single Node
+The graph reveals an interesting twist: Modin on a single node starts off slower than Pandas when using 4 workers or fewer. This could be because the overhead of distributing data and spinning up the Ray cluster doesn't pay off with so few workers. It's like trying to split chores among two people when you really need a full team. But here's where it gets exciting: crank up the CPU count (even just to 8), and Modin shifts into high gear, outperforming Pandas even on a single node. Talk about a comeback!
+
+<img src="imgs/blog_post_4/Modin_single_node.png" alt="Perf Results single node"  style="display: block; margin-left: auto; margin-right: auto;">
+
+
+### Performance on Scaling Nodes to 32
+
+The performance continues to soar as we add more nodes to the cluster. With a 32-node cluster, utilizing all 16 CPUs in each node, the script executes in a fraction of the time it takes on a single node. This results in a dramatic performance boost, showcasing the impressive scalability of Modin.
+
+<img  src="imgs/blog_post_4/Modin_multiple_nodes.png" alt="Perf Results multinode"  style="display: block; margin-left: auto; margin-right: auto;">
+
+#### Appendix
+
+All performance measurements for this post were made on an AWS r6a.4xlarge instance.
+
+- Modin version: 0.30.0
```

Comment:
Why not 0.31.0? We were planning to put more operations in the script, right?

Reply:
Just the docker image was not updated for running, I will rerun with 0.31.0 and update numbers.